src/main/java/org/tdl/vireo/controller/SubmissionController.java

```diff
@@ -990,9 +990,18 @@ public ApiResponse removeFile(@WeaverUser User user, @PathVariable Long submissi
             apiResponse = new ApiResponse(ERROR, "You are not allowed to delete license files!");
         } else {
             if (user.getRole().equals(Role.ROLE_ADMIN) || user.getRole().equals(Role.ROLE_MANAGER) || uri.contains(String.valueOf(hash))) {
```

Comment:
Why does manager or admin get to delete a file uploaded? That runs the risk of a student uploaded a file from wherever (laptop, cellphone, lab, home personal computer) and since uploading lost all work for unknown reasons. This scenario should be addressed and a communication is required before any file deletion!

Reply:
If it is not your account, you cannot change any interaction with Vireo. Otherwise, the application is suspicious of nefarious activity.